src/Arcus.Messaging.ServiceBus.Core/Extensions/MessageExtensions.cs

```diff
@@ -0,0 +1,23 @@
+﻿using Arcus.Messaging.Abstractions;
+using GuardNet;
+
+// ReSharper disable once CheckNamespace
+namespace Microsoft.Azure.ServiceBus
+{
+    public static class MessageExtensions
+    {
+        /// <summary>
+        ///     Gets the transaction id that is linked to this message
+        /// </summary>
+        /// <param name="message">Message to process</param>
+        /// <returns>Transaction id for message</returns>
+        public static string GetTransactionId(this Message message)
```

Comment:
Is an empty string a valid transaction ID? If not, maybe we could also make it a `Try...` with output variables.

Reply:
I think the current approach is ok